hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/BaseHeapData.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.serialization.impl;
+
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.util.HashUtil;
+
+import java.util.Arrays;
+
+import static com.hazelcast.util.JVMUtil.REFERENCE_COST_IN_BYTES;
+
+/**
+ * A {@link Data} implementation where the content lives on the heap.
+ */
+public abstract class BaseHeapData implements Data {
```

Comment:
Why not call this AbstractHeapData? Abstract is the recommended prefix for Abstract classes.